tfe/data_source_organizations.go

```diff
@@ -0,0 +1,57 @@
+package tfe
+
+import (
+	"fmt"
+	"log"
+
+	tfe "github.com/hashicorp/go-tfe"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceTFEOrganizations() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceTFEOrganizationList,
+
+		Schema: map[string]*schema.Schema{
+			"names": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
```

Comment:
I wonder if this is necessary? For workspaces, `names` is a required argument you must provide; here in organizations, you're just fetching all organizations accessible by the authenticated token. 

Since `ids` (which matches the equivalent workspaces one) includes both names and IDs, this just feels like duplication, no?

Reply:
I put `names` here as well because we use the organization's `name` as a primary identifier throughout the API, and I figured a common usage pattern is that users would query this data source and iterate through the names. That being said, one could also just iterate through `ids` (IDs/Names map) and do an action using the `name`.

Reply:
Hmmm, I was actually wondering the opposite! We use `id` [elsewhere in the provider](https://registry.terraform.io/providers/hashicorp/tfe/latest/docs/resources/organization#id) to mean the organization's name, so it seems like it could be confusing to use `id` to mean the organization's external id 😬 plus you can't really use an organization's external id for anything right now, though of course that may change in the future. 

Possibly terrible idea: what if we left `ids` out of this data source completely and had `names` as the only attribute?

Reply:
> That being said, one could also just iterate through ids (IDs/Names map) and do an action using the name.

Yeah that was my thought

The fact that we don't ever use external IDs for orgs is problematic, and I hope we make some enhancements to allow for either in the future and actually change things like the reference CJ linked, but I can also appreciate that it might be overoptimizing for now. I think `names` as the only attribute is just fine, actually! Maybe we could do that and leave the ID map for when (because we probably will) we migrate `r/tfe_organization.id` over 🤷‍♂️

Reply:
> The fact that we don't ever use external IDs for orgs is problematic

I'm in favor keeping `names`. The user is used to using the organization name to execute operations, so now seeing "ids" _only_ (and not `name`) will be confusing. But like CJ pointed, we set `id` [elsewhere](https://github.com/hashicorp/terraform-provider-tfe/blob/master/tfe/resource_tfe_organization.go#L92) as [name](https://github.com/hashicorp/terraform-provider-tfe/blob/master/tfe/resource_tfe_organization.go#L172) where its not actually the ID (external_id) but in fact is a name...😢.

I'm actually in favor of keeping both. `names` because the user is already used to using that as the primary "id". And keeping the `ids` map, even though it may not be immediately used, is good because we can put the pieces in place for a future migration to start using external_id as the primary identifier instead of names.

Reply:
Good discussion, sounds good to me!